### ISSUE TITLE
Invalid Screen / Long Profile Wait

### ISSUE THREAD

**Issue by ProdigizedDP** [EchoControl/Gunkolle#18]
**Desktop:**
What graphics card are you using?
NVIDIA GeForce GTX 1080 

**Config File:**
[Do Not Modify]
Busy=0
[Variables]
SortieInterval=1000
MaxRandomWaitS=3000
MinRandomWaitS=3000
MinRandomWait=0
MaxRandomWait=0
WINID=ahk_class Qt5QWindowIcon
PauseHr=9
PauseMn=00
ResumeHr=12
ResumemN=00
Doll1=M4A1
Doll2=G41
Doll3=ERROR
Doll4=ERROR
WeaponType=AssaultRifle
Enhancement=0
ProductionTdoll=0
ProductionEquipment=0
FriendCollector=0
BatteryCollector=0
AutoBattleResend=0
WorldSwitcher=4_3E
LastXS=1481
LastYS=891
[GPU]
AMD=0
#For AMD or others GPU's change value 0 to 1. For NVidia value must be 0

#combatsims can be of values 1,2, or 3
CombatSimsData=2

LastXS=860
LastYS=775

**Describe the bug**
When running the AHK script, it says Invalid screen. I'm using the default background. This started today when an update hit, seems like there have been a few UI changes, I'm not too sure how that's impacted the script seeing the screen. Another bug I've run into before the update, it would go into formation, filter, and stop at Profile Wait. It would stay at the profile wait for 1-2 minutes and then pick the girl. It would go into the second echleon after to resupply the other carry, filter, & then it would wait for 1-2 minutes at Profile Wait.

**Screenshots**

Invalid Screen
![Screenshot1](https://user-images.githubusercontent.com/14143426/67339354-ea760c00-f4df-11e9-93e0-97fed82ec691.png)

Invalid Screen gif
![Invalid Screen](https://user-images.githubusercontent.com/14143426/67339381-ffeb3600-f4df-11e9-8c2b-fdc354e482dd.gif)

ProfileWait gif
![ProfileWait](https://user-images.githubusercontent.com/14143426/67339408-0da0bb80-f4e0-11e9-93bc-0cdaffe10c54.gif)

**Comment by ReversalGM** [comment]
LOL the whole script obviously broke because of the idiotic client update. all the maps are broken and pretty much all functionality is dead. Don't have time to fix it right now but I will do it in a week or so. Maybe someone can fix it sooner. Pretty much everything needs to be updated. Also the map zoom is messed up too so it looks like the script needs to be responsible for zooming as well.

Also glad I'm not the only one who got the profile waiting for 2 minute one day all of a sudden. I just deleted something redundant and it is fixed.

**Comment by ProdigizedDP** [comment]
Looking forward to seeing any work you put out then, & yeah not sure what I would have deleted to fix the profile wait. I just woke up the next day and it was taking forever to corpse drag. Although that can probably use some tweaking as well considering the tdoll with no ammo will just swap into the other echleon when you pick the other now.

**Comment by Dustio** [comment]
Any ideas on how to remake the maps? FindClick seems to be mostly unusable because the fog spread to all maps when zoomed out. I've tried ClickM from Mouse.ahk but it doesn't work half the time.

**Comment by EchoControl** [comment]
Hey guys. 

Happy to see someone still using it. I'm planing on updating this soon, after fixing some core issues. Also, I would love some assistance, if some of you are willing to help, please text me on discord or telegram.

I have some ideas on how to fix zoom feature and other issues.

**Comment by EchoControl** [comment]
Also, Invalid Screen issue seems to happen also on nvidia cards and this part should be rewritten or fixed. I think about replacing it with other method.

**Comment by yocool13** [comment]
I'd love to help get it working, but I sadly have no clue how any of this works. I could provide testing on an AMD card I guess.

**Comment by EchoControl** [comment]
That's cool! Firstly today I'll focus on invalid screen issue and then move on implementing new method for choosing amd or nvidia cards. Hopefully will be done till Friday, but can't guarantee.

**Comment by Dustio** [comment]
I don't have a lot of experience with code but I can understand enough to make maps and a few other changes. Some things I've done were switched over to LDPlayer and added gui checkboxes for more t-doll selection (scrolling down to more 5/4 stars). On LDPlayer zooming in and out is a little different than Nox, not sure how much of a difference it could make.

I tried to look at the invalid screen issue but don't know how pixelsearch or choosing new cards works at all lol.

**Comment by BuffaloFrail** [comment]
watching with bated breath..

**Comment by EchoControl** [comment]
I'm kinda stuck atm with zoomout feature not working, trying to come up with some alternative for it

**Comment by Malinalxochitl** [comment]
Does it work with logistics yet? Unless I'm reading the code wrong, you'd only need to update the reference pixels to get that to work.

**Comment by EchoControl** [comment]
Yeah, I guess. Haven't looked at it atm, check WIP branch

**Comment by ReversalGM** [comment]
The break started so finally could work on a fix. The invalid window problem has been solved. What happened was update removed some reference pixels on the main screen, resulting in the error. Could just delete the pixel check, or replace it with pixel colours that does exist. The rest should be just replacing images.

However, got stuck on the map zoom. Zooming all the way out and letting it reset would result in a bunch of clouds flying over the images... not ideal. So I tried to just drag the map without zooming out. This revealed a new problem: the mouse.ahk drag function is imprecise, dragging a different amount each time, even after I removed the offset for humanization. This meant the end result would be different each time causing the final images not to be found. Previously this was not a problem as the boundaries of the map were much smaller and you could just drag an excessive amount and it will create a consistent result every time by hitting the boundary. Let me know if you guys are experiencing this as well.

I could create a temp-fix using only location and no more image scanning and rely only on sleep timers, but this is of course not ideal. Going to explore more options tomorrow. Will try to produce an expedition fix at minimum.

**Comment by Dustio** [comment]
I uploaded my LDPlayer version with no pixelsearch and made some maps using janky ClickS and sleep methods.  Start all maps zoomed out if anyone wants to test. There are still issues that I found while testing.

A big one that hasn't been mentioned is that nodes() is broken, and I'm not sure how to fix it so I did a slow workaround with sleeps. 
When coming back to home, it can only click on one expedition returning at a time, and freezes up when another arrives.
Doll replacement can mess up the formation, but I think this is a bug with the game and will hopefully be fixed.
I'm sure there is more stuff that needs fixing but I can't remember them atm lol.

I added some new maps, 4_6data, 6_3N, and 10_4E. 
There are also 2 new sort buttons, which when selected, scroll down to more t-dolls for selection (1 scrolls a little, 2 scrolls more). 

If there's anything missing/not working let me know. Also I might try to integrate this into Echo's Gunkolle later so that it can work with Nox too.

**Comment by yocool13** [comment]
I've been trying out your LDPlayer version and it seems to work well. After it having completed runs until M16A1 was critically damaged it seemed to not try to repair her. Wondering if you also experience this.

**Comment by Dustio** [comment]
Yep, was using an old pic for some reason, should be working now I think.

**Comment by yocool13** [comment]
Yep, it seems to work quite well now, thank you very much!

**Comment by BuffaloFrail** [comment]
Hey Dustio, your script seems to be working on AMD also, just had to change a few pictures out (Resupply and Execute Plan), just thought you should know. Also the script does not skip the post-battle animations, but from what I can tell this is intentional, as a workaround.

**Comment by Dustio** [comment]
Cool, I guess it makes sense that some pics wouldn't work since I'm using nvidia. Also guess I should've stated that I use the uncensored t-doll recipe. I've tried fixing the battle skip stuff but can't get it to work consistently. I can't seem to find a fix for multiple logistic expeditions returning during transition either; I think it's getting caught on counter=0 or something.

**Comment by sjk0603** [comment]
Hello @Dustio , I'm excited that you're keeping this project alive and I greatly appreciate your work! I downloaded latest LD player and it seems like I'm getting this error. I tested it by making a new image to search for the formations button and it seems to work again. Is there a reason why the old images aren't working for me? I'm using a gtx 1060 so I doubt it's a graphics card issue.
![error](https://user-images.githubusercontent.com/43340980/70054010-ae38cf80-15a4-11ea-9b9d-a61089c4c866.png)

**Comment by Dustio** [comment]
Thanks! I haven't switched over to the newest Girls Frontline client yet, and I don't think I will until Mica fixes the bugs or forces an update. You can probably find an older apk version of the game somewhere. If that isn't the issue, then I'm not sure what could be causing the problem.

**Comment by sjk0603** [comment]
Ah, I didn't know I could still use the older version! Thank you, I'll try that and report back when I can.

**Comment by ProdigizedDP** [comment]
@Dustio It would be pretty epic if you could open an issues on your repo for Gunkolle since I feel like it's a separate topic from the issue I originally opened. But since you don't have one currently opened I'll ask here. I've been using your LDplayer version for a while now and haven't run into any issues aside from the singularity event map 3-1, the map you have for five seven. Seems like sleeps may be off or something, i'm not sure. Might be on my side so I wanted to confirm with you that it's working fine for you?

**Comment by Dustio** [comment]
Oh my bad, it's open now. As for the map, I had over 100 attempts without a problem, where exactly in the run does it go wrong for you? I can only guess and add a small sleep to where something might go wrong.
